src/application/service/UpdateUser.ts

```diff
@@ -10,14 +10,15 @@ import { Address } from '../../domain/model/user/Address';
 import { Sex } from '../../domain/model/user/Sex';
 import { Country } from '../../domain/model/user/Country';
 import { DateOfBirth } from '../../domain/model/user/DateOfBirth';
+import { UserNotFoundError } from '../../domain/model/user/UserNotFoundError';
 
 export class UpdateUser {
   constructor(private userRepository: UserRepository) {}
 
   public async execute(id: string, command: UpdateUserCommand) {
     const existingUser = await this.userRepository.findByUserId(new UserId(id));
     if (!existingUser) {
-      throw new UserNotFoundError(id);
+      throw new UserNotFoundError(new UserId(id));
```

Comment:
In `src/api/routes/user.ts` you stopped using this Error class, instead using the more generic `ResourceNotFoundError`.  Is there any reason for them to be different?

Reply:
`ResourceNotFound` is a parent of several `*NotFound` errors. This way we can handle all the family in the same way (if we want to). In some cases we might still want to deal with them differently. For example a user not being found when the ID is in the path should lead to `404` but when the not found resource is in the body it should lead to `422`

Reply:
In fact I see I am not using it properly here. It should be:

```
  if (error instanceof UserNotFoundError) {
    throw new ApiError(422, 'user.not-found');
  }
  if (error instanceof ResourceNotFoundError) {
    throw new ApiError(422, `${error.resourceName}.not-found`);
  }
  if (error instanceof DomainValidationError) {
    throw new ApiError(422, `user.invalid.${error.field}`, error.reason);
  }
  throw error;
```